### ISSUE TITLE
Asset Recompile API Endpoint

### ISSUE THREAD

**Issue by mbiang** [sensu/bonsai-asset-index#71]
When an Github release's assets are changed, Github doesn't issue a webhook event that we can subscribe to to update the release assets on Bonsai.   Bonsai currently supports a "Recomplile"  function which can be manually initiated by the asset owner via the site UI.  

It would be nice to have a API endpoint which could be called from a CI pipeline to initiate a recompile programmatically.

**Expected Behavior**

- An API endpoint should exist that allows a site administrator or asset owner to recompile an asset.
- This endpoint should be fully documented in the API documentation
- This endpoint should be only function for authenticated users (via API auth), and authorized users (site owner, admin).

**Current Behavior**

- None of the above functionality exists currently.

**Comment by pzupan** [comment]
Created Pull request #84

**Comment by mbiang** [comment]
This issue is dependent on #85.
